main.cpp

```diff
@@ -52,7 +52,39 @@ static cell AMX_NATIVE_CALL n_HelloWorld_CheckArgsTest(AMX *amx, cell *params)
 	logprintf("This line shouldn't be printed");
 	return 1;
 }
+static cell AMX_NATIVE_CALL n_HelloWorld_SetAmxString(AMX* amx, cell* params)
+{
+	enum
+	{
+		args_size,
+		arg_c_string,
+		arg_cpp_string,
+		__dummy_elem_, num_args_expected = __dummy_elem_ - 1
+	};
+	if (!CheckArgs())
+		return 0;
+	
+	const char* c_str = "Test C-string";
+	std::string cpp_str = "Test cplusplus-string";
 
+	pluginutils::SetAmxString(amx, params[arg_c_string], c_str);
+	pluginutils::SetAmxString(amx, params[arg_cpp_string], cpp_str);
+	return 1;
+}
+static cell AMX_NATIVE_CALL n_Test_PrintString(AMX* amx, cell* params)
+{
+	enum
+	{
+		args_size,
+		arg_string,
+		__dummy_elem_, num_args_expected = __dummy_elem_ - 1
+	};
+	if (!CheckArgs())
+		return 0;
+	std::string param_str = pluginutils::GetAmxString(amx, params[arg_string]);
+	logprintf("%s: %s", pluginutils::GetCurrentNativeFunctionName(amx), param_str.data());
```

Comment:
Before C++11 the array returned by `data()` is not required (and thus not guaranteed) to be null-terminated. You'll probably want to use `c_str()` instead.

Reply:
Oh, no. "The returned array is null-terminated, that is, data() and c_str() perform the same function."

Link: https://en.cppreference.com/w/cpp/string/basic_string/data

Reply:
This plugin template uses the C++03 standard (at least that's the intention), so the user could choose between that one, C++11 or C++17, without being forced to use any of the newer standards, and I'm not planning to change this.

Reply:
You're referring to the "since C++11" section in that article, while I said the problem persists **before C++11**, because, as I mentioned earlier, this plugin template uses C++03.